docs/src/model_setup/boundary_conditions.md

```diff
@@ -1,62 +1,173 @@
+├── east: PeriodicBoundaryCondition
+├── south: ValueBoundaryCondition: 0
+├── north: ValueBoundaryCondition: 0
+├── bottom: ValueBoundaryCondition: 0
+├── top: ValueBoundaryCondition: 0
+└── immersed: FluxBoundaryCondition: Nothing
+```
+
+Boundary conditions are passed to `FieldBoundaryCondition` to build boundary conditions for each
+field individually, and then onto the model constructor (here `NonhydrotaticModel`) via the 
+keyword argument `boundary_conditions`.
+The model constructor then "interprets" the input and builds appropriate boundary conditions
+for the grid `topology`, given the user-specified `no_slip` default boundary condition for `Bounded`
+directions. In the above example, note that the `west` and `east` boundary conditions are `PeriodicBoundaryCondition`
+because the `x`-topology of the grid is `Periodic`.
+
+### Example: specifying boundary conditions on individual boundaries
+
+To specify no-slip boundary conditions on every `Bounded` direction _except_
+the surface, we write
+
+```jldoctest bcsintro
+julia> free_slip_surface_bcs = FieldBoundaryConditions(no_slip_bc, top=FluxBoundaryCondition(nothing));
+
+julia> model = NonhydrostaticModel(; grid, boundary_conditions=(u=free_slip_surface_bcs, v=free_slip_surface_bcs, w=no_slip_field_bcs));
+
+julia> model.velocities.u.boundary_conditions
+Oceananigans.FieldBoundaryConditions, with boundary conditions
+├── west: PeriodicBoundaryCondition
+├── east: PeriodicBoundaryCondition
+├── south: ValueBoundaryCondition: 0
+├── north: ValueBoundaryCondition: 0
+├── bottom: ValueBoundaryCondition: 0
+├── top: FluxBoundaryCondition: Nothing
+└── immersed: FluxBoundaryCondition: Nothing
+
+julia> model.velocities.v.boundary_conditions
+Oceananigans.FieldBoundaryConditions, with boundary conditions
+├── west: PeriodicBoundaryCondition
+├── east: PeriodicBoundaryCondition
+├── south: OpenBoundaryCondition: Nothing
+├── north: OpenBoundaryCondition: Nothing
+├── bottom: ValueBoundaryCondition: 0
+├── top: FluxBoundaryCondition: Nothing
+└── immersed: FluxBoundaryCondition: Nothing
+```
+
+Now both `u` and `v` have `FluxBoundaryCondition(nothing)` at the `top` boundary, which is `Oceananigans` lingo
+for "no-flux boundary condition".
+
 ## Boundary condition classifications
 
-1. [`Periodic`](@ref)
-2. [`Flux`](@ref)
-3. [`Value`](@ref) (Dirchlet)
-4. [`Gradient`](@ref) (Neumann)
-5. [`Open`](@ref)
+There are three primary boundary condition classifications:
 
-Boundary conditions are constructed using the classification as a prefix: `FluxBoundaryCondition`, `ValueBoundaryCondition`, and so on.
+1. [`FluxBoundaryCondition`](@ref) specifies fluxes directly.
 
-## Starter tips
+   For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near-surface fluid.
+   If there is a known `diffusivity`, you can express `FluxBoundaryCondition(flux)`
+   using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).
```

Comment:
I like this explanation using `GradientBoundaryCondition`, but you haven't introduced `GradBoundaryCondition` yet. Maybe you could keep the text but the order could be value, gradient, flux? This also has the advantage of being in increasing order of complexity IMO.

Reply:
Ok, I moved this text into `GradientBoundaryCondition`.